docs/api/response-types.md

```diff
@@ -0,0 +1,302 @@
+        return handle_dataframe(response.to_dataframe())
+    
+    # Check for visualization
+    if hasattr(response, 'dataset_id'):
+        return handle_visualization(response)
+    
+    # Check for text
+    if hasattr(response, 'content'):
+        return handle_text(response.content)
+```
+
+### Method 2: Type Property
+```python
+def handle_response_by_type(response):
+    response_handlers = {
+        'DfElement': handle_dataframe,
+        'GraphElement': handle_graph,
+        'TextElement': handle_text,
+        'ExceptionElement': handle_error,
+        'KeplerElement': handle_map,
+        'PerspectiveElement': handle_chart
+    }
+    
+    handler = response_handlers.get(response.type, handle_unknown)
+    return handler(response)
+```
+
+### Method 3: Try-Except Pattern
+```python
+def safe_handle_response(response):
+    try:
+        # Try DataFrame first (most common)
+        df = response.to_dataframe()
+        if df is not None:
+            return process_dataframe(df)
+    except AttributeError:
+        pass
+    
+    try:
+        # Try text content
+        if response.content:
+            return process_text(response.content)
+    except AttributeError:
+        pass
+    
+    # Handle other types...
+```
+
+## Advanced Response Handling
+
+### Streaming Responses
+For long-running queries, responses may be streamed:
+
+```python
+response = client.add_cell(thread.id, "Analyze 10M records...")
+
+# Check status
+while response.status == "running":
+    print(f"Progress: {response.progress}%")
+    time.sleep(1)
+    response = client.get_cell_status(thread.id, response.cell_id)
+```
+
+### Caching Responses
+For expensive queries, cache responses:
+
+```python
+import pickle
+
+def cached_query(client, thread_id, query, cache_file):
+    try:
+        # Try to load from cache
+        with open(cache_file, 'rb') as f:
+            return pickle.load(f)
+    except FileNotFoundError:
+        # Run query and cache result
+        response = client.add_cell(thread_id, query)
+        with open(cache_file, 'wb') as f:
+            pickle.dump(response, f)
+        return response
+```
+
+### Custom Response Wrappers
+Create your own response wrapper for easier handling:
+
+```python
+class LouieResponse:
+    def __init__(self, raw_response):
+        self.raw = raw_response
+        self._dataframe = None
+        self._visualizations = []
+        self._insights = []
+        self._parse_response()
+    
+    def _parse_response(self):
+        # Parse different element types
+        if hasattr(self.raw, 'elements'):
+            for element in self.raw.elements:
+                if element.type == 'DfElement':
+                    self._dataframe = element.to_dataframe()
+                elif element.type in ['GraphElement', 'KeplerElement']:
+                    self._visualizations.append(element)
+                elif element.type == 'TextElement':
+                    self._insights.append(element.text)
```

Comment:
hard-coding types is risky, should point to a file or generate or some other docs file with these